containers/mudata-to-ui/context/main.py

```diff
@@ -0,0 +1,133 @@
+        mdata.mod['rna'].obs['leiden_wnn'] = mdata.obs['leiden_wnn']
+        if has_cbb:
+            mdata.mod['rna'].obs['cluster_cbb'] = mdata.mod['atac_cbb'].obs['cluster']
+        mdata.mod['rna'].obs['leiden_rna'] = mdata.mod['rna'].obs['leiden']
+
+        # Tuples of column name, display name, and modality name prefixes
+        cluster_columns = [
+            ["leiden_wnn", "Leiden (Weighted Nearest Neighbor)", "wnn"],
+            ["leiden_cbg", "Leiden (ATAC Cell x Gene)", "cbg"],
+            ["leiden_rna", "Leiden (RNA)", "rna"],
+            ["cluster_cbb", "Cluster (ATAC Cell x Bin)", "cbb"] if has_cbb else None,
+        ]
+        cluster_columns = [col for col in cluster_columns if col is not None]
+
+        # Convert sparse layer matrices to CSC format for performance
+        for modality in [rna, cbg]:
+            for layer in modality.layers:
+                if isinstance(modality.layers[layer], sparse.spmatrix):
+                    modality.layers[layer] = modality.layers[layer].tocsc()
+
+        # If the main matrix is sparse, it's best for performance to
+        # use non-sparse formats to keep the portal responsive.
+        # In the future, we should be able to use CSC sparse data natively
+        # and get equal performance:
+        # https://github.com/theislab/anndata/issues/524 
+        for data_layer in [rna, cbg, mdata]:
+            if isinstance(data_layer.X, sparse.spmatrix):
+                data_layer.X = data_layer.X.todense()
+
+        # It is now possible for adata.X to be empty and have shape (0, 0)
+        # so we need to check for that here, otherwise there will
+        # be a division by zero error during adata.write_zarr
+        # Reference: https://github.com/hubmapconsortium/salmon-rnaseq/blob/dfb0e2a/bin/analysis/scvelo_analysis.py#L69
+        chunks = (mdata.shape[0], VAR_CHUNK_SIZE) if mdata.shape[1] >= VAR_CHUNK_SIZE else None
+        
+        zarr_path = output_dir / (Path(h5mu_file).stem + ".zarr")
+        mdata.write_zarr(zarr_path, chunks=chunks)
+
+        if has_cbb:
+            # Create interval column if it is not present in the original data
+            if "interval" not in cbb.var:
+                cbb.var["interval"] = cbb.var.index
+                cbb.var["interval"] = cbb.var.apply(lambda row: f"{row['chrom']}:{row['bin_start']}-{row['bin_stop']}", axis="columns")
+
+            # Write multivec zarr files for each clustering
+            for column, name, mod in cluster_columns:
+                adata_to_multivec_zarr(cbb, 
+                        f"{mod}.multivec.zarr", 
+                        obs_set_col=column, obs_set_name=name, 
+                        obs_set_vals=None, var_interval_col="interval", 
+                        # NOTE: Currently using grch37 as the assembly as further info has not yet been provided
+                        layer_key=None, assembly="grch37", starting_resolution=5000)
```

Comment:
I think we just need to clarify whether the `chr` prefix needs to be added in the interval names so that they match the `negspy` information for the `assembly` that we specify as being used upstream. https://github.com/pkerpedjiev/negspy/blob/master/negspy/data/hg38/chromInfo.txt

Reply:
We just had a follow-up discussion - it seems that the data is `hg38`, but the `ArchR` output doesn't include the `chr` prefixes. My planned solution to this issue is to include the `chr` prefix as part of the processing here with a heuristic check to make sure that the first element doesn't start with `chr` already.